x/migration/keeper/msg_server_claim_morse_supplier_test.go

```diff
@@ -1,3 +1,310 @@
+		ClaimedSupplierStake: morseClaimableAccount.GetSupplierStake(),
+		ClaimedBalance: expectedClaimedUnstakedTokens.
+			Add(morseClaimableAccount.GetApplicationStake()),
+		SessionEndHeight: expectedSessionEndHeight,
+		Supplier:         &expectedSupplier,
+	}
+	require.Equal(t, expectedRes, msgClaimRes)
+
+	// Assert that the persisted MorseClaimableAccount is updated.
+	expectedMorseAccount := morseClaimableAccount
+	expectedMorseAccount.ShannonDestAddress = shannonDestAddr
+	expectedMorseAccount.ClaimedAtHeight = ctx.BlockHeight()
+	foundMorseAccount, found := k.GetMorseClaimableAccount(ctx, msgClaim.MorseSrcAddress)
+	require.True(t, found)
+	require.Equal(t, *expectedMorseAccount, foundMorseAccount)
+
+	// Assert that an event is emitted for each claim.
+	expectedEvent := &migrationtypes.EventMorseSupplierClaimed{
+		MorseSrcAddress:      msgClaim.MorseSrcAddress,
+		ClaimedBalance:       expectedClaimedUnstakedTokens,
+		ClaimedSupplierStake: supplierStake,
+		SessionEndHeight:     expectedSessionEndHeight,
+		Supplier:             &expectedSupplier,
+	}
+	claimEvents := events.FilterEvents[*migrationtypes.EventMorseSupplierClaimed](t, ctx.EventManager().Events())
+	require.Equal(t, 1, len(claimEvents))
+	require.Equal(t, expectedEvent, claimEvents[0])
+}
+
+func TestMsgServer_ClaimMorseSupplier_Error(t *testing.T) {
+	claimableUnstakedBalance := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 1000)
+	claimableSupplierStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 200)
+
+	k, ctx := keepertest.MigrationKeeper(t)
+	srv := keeper.NewMsgServerImpl(k)
+
+	morsePrivKey := testmigration.GenMorsePrivateKey(t, 0)
+	morseClaimableAccount := &migrationtypes.MorseClaimableAccount{
+		MorseSrcAddress:  sample.MorseAddressHex(),
+		PublicKey:        morsePrivKey.PubKey().Bytes(),
+		UnstakedBalance:  claimableUnstakedBalance,
+		ApplicationStake: cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 0),
+		SupplierStake:    claimableSupplierStake,
+		// ShannonDestAddress: (intentionally omitted),
+		// ClaimedAtHeight:    (intentionally omitted),
+	}
+
+	accountState := &migrationtypes.MorseAccountState{
+		Accounts: []*migrationtypes.MorseClaimableAccount{
+			morseClaimableAccount,
+		},
+	}
+	accountStateHash, err := accountState.GetHash()
+	require.NoError(t, err)
+
+	_, err = srv.ImportMorseClaimableAccounts(ctx, &migrationtypes.MsgImportMorseClaimableAccounts{
+		Authority:             authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		MorseAccountState:     *accountState,
+		MorseAccountStateHash: accountStateHash,
+	})
+	require.NoError(t, err)
+
+	// Claim the MorseClaimableAccount with random Shannon owner & operator addresses.
+	msgClaim, err := migrationtypes.NewMsgClaimMorseSupplier(
+		sample.AccAddress(),
+		sample.AccAddress(),
+		accountState.Accounts[0].GetMorseSrcAddress(),
+		morsePrivKey,
+		testSupplierServices,
+	)
+	require.NoError(t, err)
+
+	t.Run("invalid claim msg", func(t *testing.T) {
```

Comment:
At a high-level (I might be wrong), a couple test cases I think we could add are:

- Trying to claim as a supplier if you had zero supplier stake prior
- Trying to claim as a supplier if you have non-zero application stake prior